bigtable/google/cloud/bigtable/table.py

```diff
@@ -412,13 +426,14 @@ def __call__(self, retry=DEFAULT_RETRY):
                   sent. These will be in the same order as the ``rows``.
         """
         try:
-            retry(self.__class__._do_mutate_retryable_rows)(self)
+            retry(self._do_mutate_retryable_rows)()
```

Comment:
@justinseinlin Did you have a specific reason for doing it in this hacky way? Or just "if it works, it sticks"?

Also, IIUC @jonparrott would deem `retry=None` as an acceptable input here, so ISTM the code should be:

```python
if retry is None:
    self._do_mutate_retryable_rows()
else:
    retry(self._do_mutate_retryable_rows)()
```

Reply:
Yes, we should change it into w/ and w/o the retry. Thanks!

Reply:
@justinseinlin Do you mind making that change in a separate PR. That will need updated logic with new tests (i.e. it's not a trivial change).

In that PR, you can discuss with @jonparrott if raising a "fake" `ServiceUnavailable` is the "right" way to force a retry from within `_do_mutate_retryable_rows()`.

Reply:
Sure. I will make a separate PR with that change.

Reply:
Since this PR also affects the same line, I'll wait for this PR to land and start from that.

Reply:
SGTM

Reply:
I have the first draft of the change ready. Will do rebase after this PR is merged, and then create a new PR. 

I'll discuss with @jonparrott on the fake error thrown in order to trigger retry. Will look into making an internal exception and updating the predicate of `DEFAULT_RETRY`.